### ISSUE TITLE
[Portal 2] Failure to recognise controller input on Steam Deck

### ISSUE THREAD

**Issue by KrisCatboy** [ValveSoftware/portal2#404]
On first startup, the game worked fine, but when I opened the controller settings to change the configuration, the game froze. (but still opened the Steam Input menu)
After forcing a shutdown and running the game again, it seems to completely ignore Steam Input and does not respond to any controller inputs (keyboard inputs work fine)

This happened when I was on the beta channel, but switching to Stable did not fix it. Neither did reinstalling the game, verifying game files, or making changes to the input configuration.

I'm guessing there is a cvar which has been incorrectly set and is therefore breaking Steam Input, I don't know what it would be, though, as I cannot type into the in-game console for some reason (On-screen keyboard inputs do nothing)

**Comment by mvittiglio** [comment]
My son and I are on the stable channel (updated to latest to date) and have run into this same issue. 

I'm still testing this out to see if it works twice but I had to load the game on my PC and then disable steam input, restart the game, then re-enable steam input. I'll update again if this works twice.

**Comment by mvittiglio** [comment]
Yep, that seemed to do it. Strangely though, his controls remain "modern" but mine look like old xbox 360 era graphics. A win is a win though so I hope this helps as a workaround.

**Comment by 0xlia** [comment]
I have the same problem and have not found a solution yet.
The portal 2 controller settings are empty :/

**Comment by superseance** [comment]
> I have the same problem and have not found a solution yet. The portal 2 controller settings are empty :/

I had this empty broken menu issue too. I went into Properties > betas > and selected previous version and it's working fine after this. the menu is back and is functional. I switched back to current version and it broke again. 

I don't know what it must have been about this patch and I don't know what's actually changed. so idk.

**Comment by lauramannoni67-glitch** [comment]
Hey guys ! I was just as desperate as you were to make my controller work in the game, but I found a way to solve this issue by using this person's tutorial (that initially aims to solve coop connection issues) : 

https://steamcommunity.com/app/1255980/discussions/0/4434443737399130404/

running the game as admin worked instantly for me, my controller was recognized with no issue ! 
I hope it will work for you as well :)

Cheers
